### ISSUE TITLE
Code blocks inside markdown file

### ISSUE THREAD

**Issue by HeinrichAD** [editorconfig-checker/editorconfig-checker#351]
Is it possible to ignore the indents inside a code block of markdown file?

My problem is that I use different indent sizes for different files. I use of example an indent of 2 spaces by default but 4 spaces for markdown files (due to some 3rd party tool requirements). If I include Code examples inside the README, I would like to use my default (2 spaces instead of 4).

#### `.editorconfig`

```ini
[*]
indent_size = 2

[*.md]
indent_size = 4
```

#### `README.md`

````md
# Title

long text

- list a
    - sublist a with "normal" markdown file **indent of 4 spaces**
- list b 

## Example

```bash
command args on line one \
  args on line two  # <<-- this line is the problem
```
````

### Problem / Error

Unfortunately, an indent of 4 is expected also inside the code blocks of the markdown file.

```txt
13: Wrong amount of left-padding spaces(want multiple of 4)
```

**Comment by mstruebing** [comment]
The behavior you are describing is expected.

You could either

a) disbale the specific markdown code blocks one by one: https://github.com/editorconfig-checker/editorconfig-checker/?tab=readme-ov-file#excluding-blocks
b) set the indent size for markdown files to two as well, 4 is a valid indentation in this case as well

I don't think we should implement a special case for the case you are describing as this would open multiple can of worms and we would need to implement similar things for every programming or markdown like language.

**Comment by HeinrichAD** [comment]
Thanks for your fast response. I will use one of the two workarounds.

> I don't think we should implement a special case for the case you are describing as this would open multiple can of worms and we would need to implement similar things for every programming or markdown like language.

That's definitely true.

**Comment by l0b0** [comment]
On a related note, is there some way to force the format of a set of lines? Something like `editorconfig-checker-mock-filename my.xml` for some XML embedded in Markdown, say.
